addons/account/models/account_move.py

```diff
@@ -4299,17 +4311,38 @@ def action_post(self):
             moves_with_payments.payment_id.action_post()
         other_moves = self - moves_with_payments
         if other_moves.filtered(lambda m: m.abnormal_amount_warning or m.abnormal_date_warning):
-            return {
+            res = {
                 'name': _("Confirm Entries"),
                 'type': 'ir.actions.act_window',
                 'res_model': 'validate.account.move',
                 'view_mode': 'form',
                 'context': {'default_move_ids': other_moves.ids},
                 'target': 'new',
             }
-        if other_moves:
+        elif other_moves:
             other_moves._post(soft=False)
-        return False
+        res = False
+        # Update any linked downpayment lines
```

Comment:
Wait ... you update the SO from the invoice when posting it, is that it ? Isn't too much magic killing the magic ?

Reply:
Yes this is how it works right now for SOs and I generalized it to also apply to POs in the same way